src/document-submit.js

```diff
@@ -61,7 +61,16 @@ export function onDocumentFormSubmit_(e) {
   user.assert(target, 'form target attribute is required: %s', form);
   user.assert(target == '_blank' || target == '_top',
       'form target=%s is invalid can only be _blank or _top: %s', target, form);
-  const shouldValidate = !form.hasAttribute('novalidate');
+
+  // amp-form extension will add novalidate to all forms to manually trigger
+  // validation. In that case `novalidate` doesn't have the same meaning.
+  const isAmpFormMarked = form.classList.contains('-amp-form');
```

Comment:
I discussed this with Dima and basically he wanted to have the basic functionality of blocking submission inside `document-submit.js` and anything extra we can do inside `amp-form.js`. It does sound that we should have the polyfill stuff inside `document-submit` but I guess the expectation is that amp-form will always be loaded, this is just to make sure in the case that it fails we still block invalid submissions.